packages/react-core/src/hooks/__tests__/useGetUrl.test.ts

```diff
@@ -0,0 +1,218 @@
+import { renderHook } from '@testing-library/react-hooks';
+import * as Storage from 'aws-amplify/storage';
+
+import useGetUrl, { UseGetUrlInput } from '../useGetUrl';
+
+const getUrlSpy = jest.spyOn(Storage, 'getUrl');
+
+const url = new URL('https://amplify.s3.amazonaws.com/path/to/the/file.jpg');
+
+const onErrorMock = jest.fn();
+
+const KEY_INPUT: UseGetUrlInput = {
+  key: 'file.jpg',
+  options: { accessLevel: 'guest' },
+  onError: onErrorMock,
+};
+
+const PATH_INPUT: UseGetUrlInput = {
+  path: 'guest/file.jpg',
+  onError: onErrorMock,
+};
+
+describe('useGetUrl', () => {
+  describe('with key params', () => {
+    it('should return a Storage URL', async () => {
+      getUrlSpy.mockResolvedValue({ url, expiresAt: new Date() });
+
+      const { onError, ...getUrlParams } = KEY_INPUT;
+
+      const { result, waitForNextUpdate } = renderHook(() =>
+        useGetUrl(KEY_INPUT)
+      );
+
+      expect(getUrlSpy).toHaveBeenCalledWith(getUrlParams);
+      expect(result.current.isLoading).toBe(true);
+      expect(result.current.url).toBe(undefined);
+
+      // Next update will happen when getUrl resolves
+      await waitForNextUpdate();
+
+      expect(getUrlSpy).toHaveBeenCalledTimes(1);
+      expect(result.current.isLoading).toBe(false);
+      expect(result.current.url).toBe(url);
+
+      getUrlSpy.mockClear();
+    });
+
+    it('should invoke onStorageGetError when getUrl fails', async () => {
+      const customError = new Error('Something went wrong');
+      const { onError, ...getUrlParams } = KEY_INPUT;
```

Comment:
No need to destructure `onError` here (same for other tests). You can just reference `onErrorMock` directly in the `expect` calls

Reply:
The desctructuring here is actually to assert `getUrl` was called with `getUrlParams`

Reply:
In that case the assertion should be:
```
expect(getUrlSpy).toHaveBeenCalledWith({ key: KEY_INPUT.key, options: KEY_INPUT.options });
```

Right?

Reply:
Yeah isn't that kind of the same thing though?